cirq/google/api/v2/program.proto

```diff
@@ -198,3 +204,97 @@ message ArgFunction {
   // The arguments to the function.
   repeated Arg args = 2;
 }
+
+// An operation that applies a modified version of a reference circuit. The
+// circuit is stored in the top-level Constants table; the mappings in this
+// object specify how that circuit should be modified for this operation.
+//
+// Multiple CircuitOperations may reference the same base circuit even if their
+// mappings of that circuit are different.
+message CircuitOperation {
+  // The index of the circuit in the top-level constant table.
+  int32 circuit_constant_index = 1;
+
+  // Specifier for repetitions of the circuit, which contains either a number
+  // of repetitions or a list of repetition IDs.
+  RepetitionSpecification repetition_specification = 2;
+
+  // Map from qubits in the "inner" circuit (referenced by
```

Comment:
I am not sure I understand why it is called inner and outer circuit, but which one is which is clarified in the comment, so I guess it is fine.